dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+     * be used in the new intermediate table.
+     * @return
+     */
+    private String insertActionsAndSteps() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieves the relationships between workflow actions and the workflow steps where they can
+     * be used.
+     * @return
+     */
+    private String selectActionsAndSteps() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_SELECT_ACTIONS_AND_STEPS;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Adds the new {@code scheme_id} column to the {@code workflow_action} table so that it is now
+     * related directly with a workflow scheme.
+     * @return
+     */
+    private String addSchemeIdColumn() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_ADD_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_ADD_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_ADD_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_ADD_SCHEME_ID_COLUMN;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Adds the new {@code requires_checkout_option} column to the {@code workflow_action} table so that it is now
+     * multivalue instead of boolean.
+     * @return
+     */
+    private String addRequiresCheckoutOptionColumn() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_ADD_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_ADD_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_ADD_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_ADD_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Creates the new intermediate table called {@code workflow_action_step}.
+     * @return
+     */
+    private String createIntermediateTable() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)